### PR TITLE
[12.0][ADD] maintenance_request_ifs_check

### PR DESCRIPTION
**Pull request by lmiguens-solvos**
Adds 4 IFS check to maintenance request.
Adds new type of stage (ifs_checked), it's mandatory to check all IFS checks to change the request to and stage with ifs_checked.
Changed views of maintenance request:
- tree: with color green request in stage with ifs_checked 
- form: put new fields into a tab
- search: new filter ifs_checked

### CODE REVIEW

File: maintenance_request_ifs_check/models/maintenance.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        string='Free strange elements')
+    ifs_check_collected_location = fields.Boolean(
+        string='Collected location')
+    ifs_check_cleaning_area = fields.Boolean(
+        string='Cleaning the area')
+    ifs_checked = fields.Boolean(
+        string='IFS checked',
+        related='stage_id.ifs_checked',
+        store=True,
+        readonly=True)
```
Comment (dalonsod) [inline maintenance_request_ifs_check/models/maintenance.py]: I'm not sure if `store` and `readonly` attributes are needed. Is `store=True` needed for searching support? Also, `readonly=True` is the default starting from v12.

File: maintenance_request_ifs_check/models/maintenance.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    @api.constrains('stage_id')
+    def _check_ifs_checked(self):
+        for req in self:
+            if req.stage_id.ifs_checked\
+                    and (not req.ifs_check_repaired_correctly
+                         or not req.ifs_check_free_strange_elements
+                         or not req.ifs_check_collected_location
+                         or not req.ifs_check_cleaning_area):
```
Comment (dalonsod) [inline maintenance_request_ifs_check/models/maintenance.py]: For this piece of code, creating and calling for a `_ifs_checked_ok()` method (or even a computed field) should be clearer, and should be a good technique for further inheritance (e.g. for upcoming addons with new `ifs_*` fields).

File: maintenance_request_ifs_check/views/maintenance_views.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    <!-- New form view: MaintenanceRequest - added new IFS checks -->
+    <record id="hr_equipment_request_view_form" model="ir.ui.view">
+        <field name="model">maintenance.request</field>
+        <field name="inherit_id" ref="maintenance.hr_equipment_request_view_form"/>
+        <field name="priority" eval="100"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='description']" position="replace">
+                <notebook>
+                    <page string="Description">
```
Comment (dalonsod) [inline maintenance_request_ifs_check/views/maintenance_views.xml]: For better inheritance, add a name="page_description" attribute

File: maintenance_request_ifs_check/views/maintenance_views.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                        <field name="description" placeholder="Internal Note ......."/>
+                    </page>
+                    <page string="IFS checks">
```
Comment (dalonsod) [inline maintenance_request_ifs_check/views/maintenance_views.xml]: For better inheritance, add a name="page_ifs_checks" attribute

File: maintenance_request_ifs_check/views/maintenance_views.xml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            </xpath>
+        </field>
+    </record>
+
+    <!-- New search view: MaintenanceRequest - added ifs_checked -->
+    <record id="hr_equipment_request_view_search" model="ir.ui.view">
+        <field name="name">equipment.request.search</field>
+        <field name="model">maintenance.request</field>
+        <field name="inherit_id" ref="maintenance.hr_equipment_request_view_search"/>
+        <field name="sequence" eval="18"/>
```
Comment (dalonsod) [inline maintenance_request_ifs_check/views/maintenance_views.xml]: I don't know the purpose of `sequence` attribute

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only reviewed code, see comments for minor changes. Before attended, I'll functionally test them

File: maintenance_request_ifs_check/models/maintenance.py (hunk 1 above)
Comment (lmiguens-solvos) [reply maintenance_request_ifs_check/models/maintenance.py]: Ok. I'll remove store and readonly. I checked that store=True is not necessary for searching support.

File: maintenance_request_ifs_check/models/maintenance.py (hunk 2 above)
Comment (lmiguens-solvos) [reply maintenance_request_ifs_check/models/maintenance.py]: I understand. I'll think about use a computed field or not.

File: maintenance_request_ifs_check/views/maintenance_views.xml (hunk 3 above)
Comment (lmiguens-solvos) [reply maintenance_request_ifs_check/views/maintenance_views.xml]: Ok. I'll do it. Good idea.

File: maintenance_request_ifs_check/views/maintenance_views.xml (hunk 4 above)
Comment (lmiguens-solvos) [reply maintenance_request_ifs_check/views/maintenance_views.xml]: Ok. I'll do it. Good idea.

File: maintenance_request_ifs_check/views/maintenance_views.xml (hunk 5 above)
Comment (lmiguens-solvos) [reply maintenance_request_ifs_check/views/maintenance_views.xml]: I do not know either. I think I copied it from another view. I put the same sequence in my other PR [12.0][ADD] maintenance_request_time_performance#1... 
I should put 16 on both, right?

**Comment by dalonsod** [comment]
Made a few cosmetic changes and squashed commits

**Review by dalonsod — APPROVED** [review]
LGTM 👍

**Pull request merged**